src/app/service/languages-bcp47-tag.service.ts

```diff
@@ -0,0 +1,339 @@
+import { Injectable } from '@angular/core';
+import { AbstractService } from './abstract.service';
+
+export interface LanguageBCP47 {
+  code: string;
+  language: string;
+  region: string;
+  name: string;
+}
+
+const languages: LanguageBCP47[] = [
```

Comment:
instead of having an array of available languages, can we use the languages returned by the `mobile-content-api.cru.org/langauges` API? the `code` & `name` from that list should be enough to drive the dropdown and includes a bunch of smaller languages that we have content available in that are missed by this list.

If the tool-group already includes a language not returned by the `/languages` API, we could just show the code if there isn't a built-in mechanism for generating a name from the BCP-47 tag.

Reply:
For reference in Android/Java there is a system provided `Locale` object that allows us to do the following: `Locale.forLanguage(bcpCode).displayName`. I'm just not sure if an equivalent exists in TypeScript/Javascript.